http/src/main/scala/org/http4s/blaze/http/http1/server/Http1ServerCodec.scala

```diff
@@ -225,16 +227,25 @@ private final class Http1ServerCodec(maxNonBodyBytes: Int, pipeline: TailStage[B
       else doFlush()
     }
 
-    final override def close(): Future[RouteResult] = lock.synchronized {
+    final override def close(cause: Option[Throwable]): Future[RouteResult] = lock.synchronized {
```

Comment:
Can `close` possibly fail with something completely unexpected, or does it fail with a (relatively) enumerable set of exceptions?

Asking because if possible, I think `close(cause: Option[Exception])` forces me to squint a little less at this. `Throwable` encompasses way too much in my eyes, a lot of which is "your program is underfined now".

Reply:
It looks like we can refine it to `Exception`, and that seems reasonable. I think the only place that will cause issues if when someone attempts to close due to the failure half of a `Future`, but I suppose we can cross that bridge later.

Reply:
Is this argument an offshoot of "all Throwables are fatal and all Exceptions are not?"  That's essentially the original JVM intent, but we have two competing definitions that contradict that in Scala (stdlib's `NonFatal` and fs2/cats-effect's `NonFatal`).  It would be nice if we had a consistent interpretation of fatality up and down the stack.

Reply:
That's a reasonable point. I'm having a change of heart, and think we should probably move it back to `Throwable`.

Reply:
It's unfortunate that `NonFatal` is an extractor rather than a type, or that a few accepted non-fatal things extend `Error`.  What @jmcardon argues for is in line with the original intent of the JVM, but we lost our way on that before there was even a Scala.

Reply:
Alright, reverted back to `Throwable`. It's disappointing, but it is what it is.